test/onnx/test_pytorch_onnx_onnxruntime.py

```diff
@@ -109,17 +108,30 @@ def inline_flatten_list(inputs, res_list):
     return res_list
 
 
-def unpack_to_numpy(value):
+def unpack_to_numpy(values):
     value_unpacked = []
-    for value_ in value:
-        value_unpacked.extend(unpack_quantized_tensor(value_))
-    value_final = [to_numpy(v) for v in value_unpacked]
-    return value_final
-
-
-def run_ort(ort_sess, input):
-    input = unpack_to_numpy(flatten_tuples(input))
-    ort_inputs = dict((ort_sess.get_inputs()[i].name, input) for i, input in enumerate(input))
+    for value in values:
+        value_unpacked.extend(unpack_quantized_tensor(value))
+    return [to_numpy(v) for v in value_unpacked]
+
+
+def run_ort(ort_sess, inputs):
+    kw_inputs = {}
+    if inputs and isinstance(inputs[-1], dict):
+        kw_inputs = inputs[-1]
+        inputs = inputs[:-1]
+    inputs = unpack_to_numpy(flatten_tuples(inputs))
+    ort_inputs = {}
+    for input_name, input in kw_inputs.items():
+        ort_inputs[input_name] = to_numpy(input)
+    inputs = to_numpy(inputs)
+    ort_sess_inputs = ort_sess.get_inputs()
+    for i, input in enumerate(inputs):
+        input_name = ort_sess_inputs[i].name
+        if input_name in ort_inputs:
+            raise ValueError(
+                f"got too many positional inputs. inputs: {inputs}. kw_inputs: {kw_inputs}")
+        ort_inputs[input_name] = input
```

Comment:
are the above change mainly for covering extra dict input that weren't covered in previous prs, and unrelated to optional input?

edit: looking at other parts of code, looks like this is required for optional.